save-backend/src/main/kotlin/org/cqfn/save/backend/repository/TestExecutionRepository.kt

```diff
@@ -101,6 +101,14 @@ interface TestExecutionRepository : BaseEntityRepository<TestExecution>, JpaSpec
      */
     fun findByExecutionIdAndTestId(executionId: Long, testId: Long): List<TestExecution>
 
+    /**
+     * @param executionId
+     * @param agentId
+     * @return list of TestExecution's
+     */
+    @Suppress("TYPE_ALIAS")
+    fun findByExecutionIdAndAgentId(executionId: Long, agentId: Long): Optional<List<TestExecution>>
```

Comment:
```suggestion
    fun findByExecutionIdAndAgentId(executionId: Long, agentId: Long): List<TestExecution>
```
I believe, spring-data methods that return collections will never return null or Optional.empty, but rather an empty collection

Reply:
Agree, removed